File: CRUD-UI/src/app/employee.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Employee } from './employee';

@Injectable({
  providedIn: 'root'
})
export class EmployeeService {

  private baseURL = "http://localhost:8081/employees";

  // Injecting HttpClient module
  constructor(private httpClient : HttpClient) { }

  // Get all employees
  getEmployeeList() : Observable<Employee[]>{
    return this.httpClient.get<Employee[]>(`${this.baseURL}`);
  }

  // Create employee
  createEmployee(employee : Employee): Observable<Object>{
    return this.httpClient.post(`${this.baseURL}`, employee); 
  }

  // Get employee by id
  getEmployeeById(id : number) : Observable<Employee>{
    return this.httpClient.get<Employee>(`${this.baseURL}/${id}`);
  }

  // Update employee
  updateEmployee(id : number, employee : Employee) : Observable<Object>{
    return this.httpClient.put(`${this.baseURL}/${id}`, employee);
  }

  // Delete employee
  deleteEmployee(id : number) : Observable<Object>{
    return this.httpClient.delete(`${this.baseURL}/${id}`)
  }
}

File: SpringBoot CRUD/src/main/java/com/example/springbootcrud/service/EmployeeService.java
package com.example.springbootcrud.service;

import com.example.springbootcrud.model.Employee;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface EmployeeService {
    public List<Employee> getEmployees();
    public Employee createEmployee(Employee employee);
    public Employee updateEmployee(Long id, Employee employee);
    public Employee getEmployeeById(Long id);
    public ResponseEntity<String> deleteEmployee(Long id);
}
